### ISSUE TITLE
When VHOST file is missing, generated nginx.conf is missing server

### ISSUE THREAD

**Issue by jeffrafter** [progrium/dokku#1575]
Somehow I got into a state where my app had no VHOST file associated with it. When I would push to deploy it would create an nginx.conf that was only

```
upstream admin.example.com { 
  server 172.17.1.49:5000;
}
```

Ultimately I had to fixed this by doing a manual 

`dokku domains:add admin.example.com admin.example.com`

Then 

`dokku ps:rebuild admin.example.com`

I was chatting with `savant` on IRC and he suggested I log a user experience issue. Happy to help recreate or give any more info.

I think this might be related to https://github.com/progrium/dokku/issues/1403

**Comment by josegonzalez** [comment]
Note: a first failed deploy - due to a broken buildpack - caused there to not be a VHOST file.

**Comment by michaelshobbs** [comment]
How did a broken buildpack affect the host filesystem?

**Comment by josegonzalez** [comment]
Broken buildpack caused a deploy failure, but the app was created. I believe we skip a step if the app exists and we are deploying.

**Comment by michaelshobbs** [comment]
so perhaps the solution here is to prescribe `apps:destroy` if the first deploy failed?

**Comment by josegonzalez** [comment]
I don't think so. I would hate to configure an app and then destroy all of that just because my deploy failed.

**Comment by michaelshobbs** [comment]
I was just referring to the **first** deployment failing

**Comment by josegonzalez** [comment]
Right. I typically configure an application environment/links and *then* deploy.

**Comment by michaelshobbs** [comment]
Is there a proposal for a solution?

**Comment by josegonzalez** [comment]
I think we should create the VHOST file if it doesn't exist and NO_VHOST is not set to 1 (or whatever other cases would cause there not to be a vhost). That *should* resolve this issue, though I'm just taking a guess.

**Comment by jeffrafter** [comment]
Another possible solution would be to have a default site that is overridden in the case of a successful configuration. Without a successful configuration it would explain the solution...

**Comment by michaelshobbs** [comment]
I think this has been fixed by 804c822ace560ed87b5e212ddee43a420cd272e2 perhaps. At least the 'empty' config won't happen anymore.
